internal/contour/listener.go

```diff
@@ -175,6 +179,19 @@ func (lvc *ListenerVisitorConfig) newSecureAccessLog() []*envoy_api_v2_accesslog
 	}
 }
 
+// requestTimeout sets any durations in lvc.RequestTimeout <0 to 0 so that Envoy ends up with a positive duration.
+// for the request_timeout value we are passing, there are only two valid values:
+// 0 - disabled
+// >0 duration - the timeout.
+// The value may be unset, but we always set it to 0.
+func (lvc *ListenerVisitorConfig) requestTimeout() time.Duration {
+
+	if lvc.RequestTimeout <= 0 {
```

Comment:
please write like this

if lvc.RequestTimeout < 0 {
   return 0
}
return lvc.RequestTimeout
```
rationale

0 * time.Second is still zero, but its of type time.Duration. If you want to write that time.Duration(0) is clearer. 

Use < 0 rather than <= 0 because zero is not special, we only need to care about negative numbers.

Reply:
thanks, that is great feedback. Will do.